sunspec/core/client.py

```diff
@@ -41,9 +43,106 @@ class SunSpecClientError(SunSpecError):
+             Device type. Possible values: :const:`RTU`, :const:`TCP`,
+             :const:`MAPPED`.
+
+        slave_id :
+            Modbus slave id.
+
+        name :
+            For :const:`RTU` devices, the name of the serial port such as 'com4'
+            or '/dev/tty2'. For :const:`MAPPED` devices, the name of the modbus
+            map file.
+
+        pathlist :
+            Pathlist object containing alternate paths to support files.
+
+        baudrate :
+            For :const:`RTU` devices, baud rate such as 9600 or 19200. Defaulted
+            by modbus module to 9600.
+
+        parity :
+            For :const:`RTU` devices, parity. Possible values:
+            :const:`PARITY_NONE`, :const:`PARITY_EVEN`
+            Defaulted by modbus module to :const:`PARITY_NONE`.
+
+        ipaddr :
+            For :const:`TCP` devices, device IP address.
+
+        ipport :
+            For :const:`TCP` devices, device IP port. Defaulted by modbus module
+            to 502.
+
+        tls :
+            For :const:`TCP` devices, use TLS (Modbus/TCP Security). Defaults
+            to `tls=False`.
+
+        cafile :
+            For :const:`TCP` devices, path to certificate authority (CA)
+            certificate to use for validating server certificates. Only used
+            if `tls=True`.
+
+        certfile :
+            For :const:`TCP` devices, path to client TLS certificate to use
+            for client authentication. Only used if `tls=True`.
+
+        keyfile :
+            For :const:`TCP` devices, path to client TLS key to use for
+            client authentication. Only used if `tls=True`.
 
+        insecure_skip_tls_verify :
+            Skip verification of server TLS certificate. Only used if
+            `tls=True`.
+
+        timeout :
+            Modbus request timeout in seconds. Fractional seconds are permitted
+            such as .5.
+
+        trace :
+            Enable low level trace.
+
+    Raises:
+
+        SunSpecClientError: Raised for any sunspec module error.
+
+    Attributes:
+
+        type
+            Device type. Possible values: :const:`RTU`, :const:`TCP`,
+            :const:`MAPPED`.
+
+        name
+            For :const:`RTU` devices, the name of the serial port such as 'com4'
+            or '/dev/tty2'. For :const:`MAPPED` devices, the name of the modbus
+            map file.
+
+        pathlist
+            Pathlist object containing alternate paths to support files.
+
+        slave_id
+            Modbus slave id.
+
+        modbus_device
+            Modbus device object. Object type is based on the device type.
+
+        retry_count
+            Request retry count. Currently not used.
+
+        base_addr_list
+            List of Modbus base addresses to try when scanning a device for the
+            first time.
+    """
+
+    def __init__(self, device_type, slave_id=None, name=None, pathlist=None, baudrate=None, parity=None, ipaddr=None, ipport=None,
```

Comment:
Looks like the argument order has changed for timeout. Looking through the code it looks as though this is fine as we are always specifying as a keyword argument. It also looks like many places use `slave_id` as a positional argument rather than keyword argument but its position has not changed.